### DISCUSSION TITLE
About BATCH_MULTIPLIER / loss:nan

### DISCUSSION THREAD

**Discussion by little-spoon** [obss/sahi#719] [Q&A] [answered]
My config:
```shell
_base_ = ["../vfnet/vfnet_r50_fpn_1x_coco.py"]

TAGS = ["vfnet", "crop=480_480", "12epochs", "num_cls=2", "repeat=5"]

DATA_ROOT = "mydataset162coco/"
#BATCH_MULTIPLIER = 3
#LR_MULTIPLIER = 1
EVAL_INTERVAL = 1
NUM_CLASSES = 2
CLASSES = ("class1", "class2")
# model settings
model = dict(
    bbox_head=dict(
        num_classes=NUM_CLASSES,
    ),
)

# dataset settings
img_norm_cfg = dict(mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type="LoadImageFromFile"),
    dict(type="LoadAnnotations", with_bbox=True),
    dict(
        type="AutoAugment",
        policies=[
            [
                dict(type="RandomCrop", crop_type="absolute_range", crop_size=(480, 480), allow_negative_crop=True),
                dict(type="Resize", img_scale=(1333, 800), keep_ratio=True),
            ],
            [
                dict(type="RandomCrop", crop_type="absolute_range", crop_size=(480, 480), allow_negative_crop=True),
                dict(type="Resize", img_scale=(1333, 800), keep_ratio=True),
            ],
            [
                dict(type="Resize", img_scale=(1333, 800), keep_ratio=True),
            ],
        ],
    ),
    dict(type="RandomFlip", flip_ratio=0.5),
    dict(type="Normalize", **img_norm_cfg),
    dict(type="Pad", size_divisor=32),
    dict(type="DefaultFormatBundle"),
    dict(type="Collect", keys=["img", "gt_bboxes", "gt_labels"]),
]
test_pipeline = [
    dict(type="LoadImageFromFile"),
    dict(
        type="MultiScaleFlipAug",
        img_scale=(1333, 800),
        flip=False,
        transforms=[
            dict(type="Resize", keep_ratio=True),
            dict(type="RandomFlip"),
            dict(type="Normalize", **img_norm_cfg),
            dict(type="Pad", size_divisor=32),
            dict(type="ImageToTensor", keys=["img"]),
            dict(type="Collect", keys=["img"]),
        ],
    ),
]

data = dict(
    samples_per_gpu=1,
    workers_per_gpu=2,
    train=dict(
        type="RepeatDataset",
        times=5,
        dataset=dict(
            type="CocoDataset",
            classes=CLASSES,
            ann_file=DATA_ROOT + "annotations/train.json",
            img_prefix=DATA_ROOT + "train/",
            pipeline=train_pipeline,
        ),
    ),
    val=dict(
        classes=CLASSES,
        ann_file=DATA_ROOT + "annotations/val.json",
        img_prefix=DATA_ROOT + "val/",
        pipeline=test_pipeline,
    ),
    test=dict(
        classes=CLASSES,
        ann_file=DATA_ROOT + "annotations/val.json",
        img_prefix=DATA_ROOT + "val/",
        pipeline=test_pipeline,
    ),
)

# optimizer
# default 8 gpu
# /8 for 1 gpu
optimizer = dict(lr=0.01/2, paramwise_cfg=dict(bias_lr_mult=2.0, bias_decay_mult=0.0))

checkpoint_config = dict(interval=1, max_keep_ckpts=1, save_optimizer=False)
evaluation = dict(interval=EVAL_INTERVAL, metric="bbox", save_best="auto")

# learning policy
lr_config = dict(policy="step", warmup="linear", warmup_iters=500, warmup_ratio=0.1, step=[8,11])

runner = dict(type="EpochBasedRunner", max_epochs=12)

# logger settings
log_config = dict(
    interval=50,
    hooks=[
        dict(type="TextLoggerHook"),
        dict(type="TensorboardLoggerHook", reset_flag=False),
    ],
)

load_from = "checkpoints/vfnet_r50_fpn_1x_coco_20201027-38db6f58.pth"
#work_dir = f"runs/visdrone/{EXP_NAME}/"
```
When I use this config to train, the result in epoch_1 is very low(mAP=0.08). Sometimes the loss=nan. 
When I use `vfnet_r50_fpn_1x_coco.py` and `fcos_r50_sahi_crop_480_960_caffe_fpn_gn-head_1x_mydataset.py`to train ,the result looks like normal.
I think there maybe something wrong with learning rate, crop or anything else.
And I am confused with `BATCH_MULTIPLIER`, what it mean? By the way , I don't have 8 gpu, I just have 2. 

Thank you for answering my question!

**Comment by fcakyon** [comment]
You should adjust the lr based on your batch size.

Standard val accuracy should be low during training, you have to perform sliced eval using the script present in the repo.

**Comment by fcakyon** [comment, marked as answer]
Check this config to see how lr is scaled. You can use lower BATCH_MULTIPLIER if you have less gpu memory.

https://github.com/fcakyon/small-object-detection-benchmark/blob/main/mmdet_configs/visdrone_vfnet/vfnet_crop_480_960_cls_60.py

**Reply by little-spoon** [reply]
Thanks a lot!

**Reply by fcakyon** [reply]
added the link to config file
